bluemira/base/parameter_frame/_parameter.py

```diff
@@ -137,12 +137,28 @@ def value(self) -> ParameterValueType:
     def value(self, new_value: ParameterValueType):
         self.set_value(new_value, source="")
 
-    def value_as(self, unit: Union[str, pint.Unit]) -> ParameterValueType:
-        """Return the current value in a given unit"""
+    def value_as(self, unit: Union[str, pint.Unit]) -> Union[ParameterValueType, None]:
+        """
+        Return the current value in a given unit
+
+        Notes
+        -----
+        If the current value of the parameter is None the function checks for
+        a valid unit conversion
+        """
         try:
             return raw_uc(self.value, self.unit, unit)
         except pint.errors.PintError as pe:
             raise ValueError("Unit conversion failed") from pe
+        except TypeError:
+            if self.value is None:
+                try:
+                    raw_uc(1, self.unit, unit)
+                    return
+                except pint.errors.PintError as pe2:
+                    raise ValueError("Unit conversion failed") from pe2
```

Comment:
I have a feeling we do something like this elsewhere. Could we refactor it into a function, maybe this?

```python
def units_compatible(unit_1: str, unit_2: str) -> bool:
```

I feel the intent of the code will be clearer this way.

Reply:
I'll factor this out either way but the only other use I can find subtly different. In the first case we avoid the error and in the second we actively want it to be thrown:

https://github.com/Fusion-Power-Plant-Framework/bluemira/blob/10c37558cc125b40150e27f82941b205da16a3df/bluemira/base/parameter_frame/_frame.py#L265

https://github.com/Fusion-Power-Plant-Framework/bluemira/blob/917683ae45da037b4cbe21f9ddc6f16d4352cd21/bluemira/base/parameter_frame/_parameter.py#L156

Reply:
The two cases I have for the factoring out of the units check are a little different one uses the pint quantity thats produced and does no conversion while the other does the check

https://github.com/Fusion-Power-Plant-Framework/bluemira/blob/917683ae45da037b4cbe21f9ddc6f16d4352cd21/bluemira/base/parameter_frame/_parameter.py#L156

https://github.com/Fusion-Power-Plant-Framework/bluemira/blob/10c37558cc125b40150e27f82941b205da16a3df/bluemira/base/parameter_frame/_frame.py#L265

I'll factor it out anyway as it is a useful function to have